goguen/formal-spec/properties.tex

```diff
@@ -0,0 +1,21 @@
+\section{Properties of the Goguen Ledger}
+\label{sec:properties}
+
+This section collects properties that the ledger described previously is expected to satisfy.
+
+\begin{itemize}
+\item The general accounting property holds with any transaction,
+  whether it is fully processed or not. This implies that the total
+  amount of Ada in the system is constant.
+\item If a transaction is accepted and marked as paying fees only
+  (i.e. $\fun{txvaltag}\, tx \in \Yes$), then the only change to the ledger
+  when processing the transaction is that the inputs marked for paying
+  fees are moved to the fee pot.
+\item If a Shelley transaction is accepted, it is fully processed.
+\item If a transaction extends the UTxO, all its scripts validate, and
+  if it has a script that does not validate, it cannot extend the
+  UTxO.
+\item A valid transaction that does not forge tokens satisfies the
+  accounting property of the Shelley ledger where the type $\Coin$ is
+  replaced by $\Value$.
+\end{itemize}
```

Comment:
(i) is true for ALL tokens, both custom and Ada. The way that the accounting property is formulated takes into consideration the `forge` field, which balances out any tokens that the transaction creates or destroys 
(v) I dont think we should refer to properties of other ledgers

(vi) all ada should be fungible with other ada
- this property is an emergent property of the ledger because it is a result of Ada not being forged, the accounting property holding for each token type, and the ledger translation of Shelley -> Goguen making only the right kind of Ada tokens, something like `(AdaPolicy, AdaTokenName, q)`

Reply:
(i) was an attempt to combine property (i) and (iii) from https://github.com/input-output-hk/cardano-ledger-specs/issues/1284
(v) I don't know about that. It would of course be a difficult property to prove (especially formally), but it provides a strong guarantee that Goguen doesn't break anything.
(vi) The issue I have here is that it isn't a property of the STS, it's a property of the STS and its initial conditions. So we could say something like: Assuming the initial state of the ledger was the result of a translation from Shelley, all tokens with `AdaPolicy` have `AdaTokenName`.

Reply:
these are all meant to be properties of a valid ledger state, derived with relevant rules from a valid initial state, so (vi) is relevant. 

(v) yea - leave it unless someone else complains

(i) i would rephrase a bit, 
"whether it is fully processed or just paying fees. In particular, this implies that the total
  amount of Ada in the system is constant."

Reply:
I've made some changes in the most recent commit to address this, so I'll mark this as resolved